packages/eas-cli/src/commandUtils/EasCommand.ts

```diff
@@ -66,4 +73,40 @@ export default abstract class EasCommand extends Command {
       setVcsClient(new GitClient());
     }
   }
+
+  private async ensureEasCliIsNotInDependenciesAsync(projectDir: string): Promise<void> {
+    let printCliVersionWarning = false;
+
+    if (await this.isEasCliInDependenciesAsync(projectDir)) {
+      printCliVersionWarning = true;
+      Log.warn(`${chalk.bold('eas-cli')} is added to the project dependencies.`);
+    }
+
+    const maybeRepoRoot = PackageManagerUtils.findWorkspaceRoot(projectDir) ?? projectDir;
+    if (maybeRepoRoot !== projectDir && (await this.isEasCliInDependenciesAsync(maybeRepoRoot))) {
+      printCliVersionWarning = true;
+      Log.warn(`${chalk.bold('eas-cli')} is added to the monorepo dependencies.`);
```

Comment:
Printing here to stdout will most likely break --json option, to avoid that you need to either
- handle JSON flag here
- log everything here to stderr.